### PR TITLE
Version id in payload

### PR DESCRIPTION
**Pull request by LogicalChaos**
Add versionId to DeleteObject, DeleteObjectsRequest and DeleteObjectError.

### CODE REVIEW

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/DeleteObjectsRequest.java [hunk 1]
```diff
@@ -89,9 +89,20 @@ public InputStream getStream() {
...
+    
+        for ( Map.Entry< String, Set< UUID > > entry : objects.entrySet() )
+        {
+            if ( null == entry.getValue() )
+            {
+                deleteObjects.add( new DeleteObject( entry.getKey(), null ) );
+            }
+            else
+            {
+                for ( UUID versionId : entry.getValue() )
```
Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/DeleteObjectsRequest.java]: Missing final

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/DeleteObjectsRequest.java [hunk 2]
```diff
@@ -89,9 +89,20 @@ public InputStream getStream() {
         final Delete delete = new Delete();
         delete.setQuiet(quiet);
         final List<DeleteObject> deleteObjects = new ArrayList<>();
-
-        for(final String objName : objects) {
-            deleteObjects.add(new DeleteObject(objName));
+    
+        for ( Map.Entry< String, Set< UUID > > entry : objects.entrySet() )
```
Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/DeleteObjectsRequest.java]: Missing final

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/DeleteObjectsRequest.java [hunk 3]
```diff
@@ -44,25 +49,20 @@
...
-        this.bucketName = bucketName;
-        
-        this.getQueryParams().put("delete", null);
-        this.objects = contentsToString(objs);
-    }
-
-    private static List<String> contentsToString(final Iterable<Contents> objs) {
-        final List<String> objKeyList = new ArrayList<>();
+    
+    private static Map< String, Set< UUID > > contentsToString( final Iterable< Contents > objs )
```
Comment (rpmoore) [inline ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/DeleteObjectsRequest.java]: You deleted the constructor this code was used in.  I believe this code is now dead.  If so, it should be removed.

File: build.gradle [hunk 4]
```diff
@@ -15,7 +15,7 @@
 
 allprojects {
     group = 'com.spectralogic.ds3'
-    version = '3.2.8-COMPAT-2'
+    version = '3.2.8-SNAPSHOT-3'
```
Comment (rpmoore) [inline build.gradle]: You should update the version number back to COMPAT.

File: ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/DeleteObjectsRequest.java (hunk 3 above)
Comment (LogicalChaos) [reply ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/DeleteObjectsRequest.java]: I wasn't sure, removed now.

**Pull request merged**